src/sentry/incidents/grouptype.py

```diff
@@ -41,6 +43,12 @@ def create_occurrence(
         data_packet: DataPacket[QuerySubscriptionUpdate],
         priority: DetectorPriorityLevel,
     ) -> tuple[DetectorOccurrence, EventData]:
+        try:
```

Comment:
🤔 this class is starting to get a bit large, should we pull it out into its own file and then import it here? That might also make it clearer with tests / what's being tested. Thoughts on something like: `incidents/metric_detector.py` then `incidents/test_metric_detector.py` for the code / tests?